app/Gists/GistClient.php

```diff
@@ -72,32 +73,43 @@ public function getGistComments($gistId)
      */
     public function postGistComment($gistId, $comment)
     {
-        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
-        $response = $this->github->getHttpClient()->post("gists/{$gistId}/comments", json_encode(['body' => $comment]));
+        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_ACCESS_TOKEN);
+        $response = $this->github->getHttpClient()->post("gists/{$gistId}/comments", [], json_encode(['body' => $comment]));
 
         return ResponseMediator::getContent($response);
     }
 
     public function starGist($gistId)
     {
-        $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
-        $this->github->getHttpClient()->put("https://api.github.com/gists/{$gistId}/star", json_encode(['body' => '']), ['Content-Length' => 0]);
+
+        if (Auth::check()) {
+            try {
+                $this->github->authenticate(Auth::user()->token, GitHubClient::AUTH_ACCESS_TOKEN);
+                $this->github->getHttpClient()->put("https://api.github.com/gists/{$gistId}/star", [], json_encode(['body' => '']), ['Content-Length' => 0]);
+                return true;
+            } catch (Throwable $e) {
+                return false;
```

Comment:
Is there any reason not to allow this to an exception if we call this method and it fails? Feels like I'd rather do that, and then turn it into whatever response I want in the controller that' calling this

Reply:
I'm actually not really sure why they are in there, would you rather me just get rid of the try catch block?

Reply:
Yep! :)